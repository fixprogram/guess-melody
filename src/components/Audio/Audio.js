import React, {Fragment} from 'react';
import PropTypes from 'prop-types';

export default class Audio extends React.Component {
  constructor(props) {
    super(props);

    this.audioRef = React.createRef();
  }

  componentDidUpdate() {
    if (this.props.isPlaying) {
      this.audioRef.current.play();
    } else {
      this.audioRef.current.pause();
    }
  }

  render() {
    const {isPlaying, src, onPlayButtonClick} = this.props;

    return (
      <Fragment>
        <button className={`track__button track__button--${isPlaying ? `pause` : `play`}`}
          type="button"
          onClick={onPlayButtonClick}
        ></button>
        <div className="track__status">
          <audio src={src} ref={this.audioRef}></audio>
        </div>
      </Fragment>
    );
  }
}

Audio.propTypes = {
  src: PropTypes.string.isRequired,
  isPlaying: PropTypes.bool.isRequired,
  onPlayButtonClick: PropTypes.func.isRequired
};
