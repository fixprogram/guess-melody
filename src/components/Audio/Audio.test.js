import React from 'react';
import renderer from 'react-test-renderer';
import Audio from './Audio';

it(`When user genre question form is not sent`, () => {
  const song = `path`;
  const tree = renderer.create(
      <Audio
        src={song}
        isPlaying={false}
        onPlayButtonClick={jest.fn()} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
}
);
