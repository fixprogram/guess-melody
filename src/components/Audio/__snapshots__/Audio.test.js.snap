// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When user genre question form is not sent 1`] = `
Array [
  <button
    className="track__button track__button--play"
    onClick={[MockFunction]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio
      src="path"
    />
  </div>,
]
`;
